src/metabase/api/dashboard.clj

```diff
@@ -539,24 +513,67 @@
                                          (assoc mapping :card-id (get dashcard-id->card-id dashcard-id)))]
     (check-parameter-mapping-permissions new-mappings)))
 
+(mu/defn ^:private classify-changes :- [:map
+                                        [:to-create [:sequential [:map [:id neg-int?]]]]
+                                        [:to-update [:sequential [:map [:id ms/PositiveInt]]]]
```

Comment:
you mean between `to-create` and others? 

because negative ids in `new-changes` is considered as `to-create` entity.

I added a docstring to this helper, hope it makes thing clearer.

Reply:
docstring added here: 0d7da010b2